src/sentry/monitors/tasks.py

```diff
@@ -390,3 +407,51 @@ def mark_checkin_timeout(checkin_id: int, ts: datetime, **kwargs):
         )
 
         mark_failed(checkin, ts=most_recent_expected_ts)
+
+
+@instrumented_task(
+    name="sentry.monitors.tasks.detect_broken_monitor_envs",
+    max_retries=0,
+    time_limit=15 * 60,
+    soft_time_limit=10 * 60,
+    record_timing=True,
+)
+def detect_broken_monitor_envs():
+    current_time = django_timezone.now()
+    open_incidents = MonitorIncident.objects.select_related("monitor").filter(
+        resolving_checkin=None,
+        starting_timestamp__lte=(current_time - timedelta(days=NUM_DAYS_BROKEN_PERIOD)),
+    )
+    for open_incident in open_incidents:
+        try:
+            organization = Organization.objects.get_from_cache(
+                id=open_incident.monitor.organization_id
+            )
+            if not features.has(
+                "organizations:crons-broken-monitor-detection", organization=organization
+            ):
+                continue
+        except Organization.DoesNotExist:
+            continue
+
+        # verify that the most recent check-ins have been failing
+        recent_checkins = (
+            MonitorCheckIn.objects.filter(monitor_environment=open_incident.monitor_environment)
+            .order_by("-date_added")
+            .values("status")[:NUM_CONSECUTIVE_BROKEN_CHECKINS]
+        )
+        if len(recent_checkins) != NUM_CONSECUTIVE_BROKEN_CHECKINS or not all(
+            [
+                checkin["status"]
+                in [CheckInStatus.ERROR, CheckInStatus.TIMEOUT, CheckInStatus.MISSED]
+                for checkin in recent_checkins
+            ]
+        ):
```

Comment:
I don't think you need the list comprehension here and can have this be a generator instead, it won't generate the whole list first and will stop once one of them is True

minor nit really, it's not like there's any real performance concern here, but worth knowing that `any` and friends can take generators, and writing a `v for v in thing` is a generator itself

```suggestion
        if len(recent_checkins) != NUM_CONSECUTIVE_BROKEN_CHECKINS or not all(
            checkin["status"]
            in [CheckInStatus.ERROR, CheckInStatus.TIMEOUT, CheckInStatus.MISSED]
            for checkin in recent_checkins
        ):
```

Reply:
oooh nah this is very good appreciate the tips here